PackageViewModel/SignPackageViewModel.cs

```diff
@@ -0,0 +1,363 @@
+            get => _password;
+            set
+            {
+                _password = value;
+                OnPropertyChanged();
+                ValidateCertificate();
+            }
+        }
+
+        public bool ShowPassword
+        {
+            get => _showPassword;
+            set
+            {
+                _showPassword = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public ICommand SelectCertificateFileCommand { get; }
+
+        public ICommand SelectCertificateStoreCommand { get; }
+
+        public ICommand ShowCertificateCommand { get; }
+
+        public bool HasError
+        {
+            get => _hasError;
+            set
+            {
+                if (_hasError != value)
+                {
+                    _hasError = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public string Status
+        {
+            get => _status;
+            set
+            {
+                if (_status != value)
+                {
+                    _status = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
+
+        public bool ShowProgress
+        {
+            get => _showProgress;
+            set
+            {
+                _showProgress = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public bool CanSign
+        {
+            get => _canSign;
+            set
+            {
+                _canSign = value;
+                OnPropertyChanged();
+            }
+        }
+
+        private void SelectCertificateFileCommandExecute()
+        {
+            if (_uiServices.OpenFileDialog("Select Certificate", "Certificate (*.pfx, *.p12)|*.pfx;*.p12|All files (*.*)|*.*", out var fileName))
+            {
+                CertificateFileName = fileName;
+            }
+        }
+
+        private void SelectCertificateStoreCommandExecute()
+        {
+            try
+            {
+                using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+                {
+                    store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+
+                    // https://github.com/NuGet/NuGet.Client/blob/adfe6d5c37834e8eb11453518e4508a534c15f8d/src/NuGet.Core/NuGet.Commands/SignCommand/SignCommandRunner.cs#L271-L283
+                    var collection = new X509Certificate2Collection();
+
+                    foreach (var certificate in store.Certificates)
+                    {
+                        if (CertificateUtility.IsValidForPurposeFast(certificate, Oids.CodeSigningEku))
+                        {
+                            collection.Add(certificate);
+                        }
+                    }
+
+                    var certificates = X509Certificate2UI.SelectFromCollection(
+                        collection,
+                        "Choose a Certificate for Package Signing",
```

Comment:
These strings (and others on the dialogs as well) should be put in resources so they can be localized

Reply:
Will do. There a currently a lot of static strings in other dialogs as well.